File: 020/Erlang/problem020.erl
-module(problem020).
-export([sum_of_digits_of_factorial/1, sum_of_digits/1]).

-import(numbers, [factorial/1]).

sum_of_digits_of_factorial(N) -> sum_of_digits(factorial(N)).

sum_of_digits(N) when N < 10 -> N;
sum_of_digits(N) -> (N rem 10) + sum_of_digits(N div 10).

File: 020/Ruby/problem020.rb
$LOAD_PATH << "../../lib/Ruby"

require 'numbers'

class Integer

  def sum_of_digits_of_factorial
    self.factorial.sum_of_digits
  end

  def sum_of_digits
    if self < 10
      self
    else
      self % 10 + (self / 10).sum_of_digits
    end
  end

end
